core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -171,11 +165,14 @@ public void doWork() {
             log.error("Failed to add record from the Kafka topic" + topic + " the local store");
           }
         }
+
+        if (commitInterval > 0 && System.currentTimeMillis() - lastCommitTime > commitInterval) {
+          log.debug("Committing offsets");
+          consumer.commitSync();
+        }
       }
-    } catch (ConsumerTimeoutException cte) {
-      // Expect ConsumerTimeout == -1, so this *should* never happen
-      throw new IllegalStateException(
-          "KafkaStoreReaderThread's ConsumerIterator timed out despite expected infinite timeout.");
+    } catch (WakeupException we) {
+      this.consumer.close();
```

Comment:
Ah, good suggestion. Originally I thought that would cause the consumer to close prematurely. But that isn't true because I'm using Long.MAX_VALUE in poll().